Makefile.in

```diff
@@ -1,185 +1,77 @@
-	@echo "========= CHECKING backend/$(ARCH) =============="
-	rm -f $(ARCH_SPECIFIC)
-	PATH=$(stage0_prefix)/bin:$$PATH \
-	  $(dune) build --root=. --profile=release --build-dir=_build1 ocamloptcomp.cma
-	rm -f $(ARCH_SPECIFIC)
-
-.PHONY: check_all_arches
-check_all_arches: ocaml-stage1-config.status stage0 \
-	config_files
-	cp ocaml-stage1-config.status ocaml/config.status
-	(cd ocaml && ./config.status)
-	if $$(grep '^ARCH64=true' ocaml/Makefile.config > /dev/null) ; \
-	then \
-		for i in $(ARCHES); do \
-		   $(MAKE) --no-print-directory check_arch ARCH=$$i || exit 1; \
-		done; \
-	else \
-		echo "Architecture tests are disabled on 32-bit platforms." ;\
-	fi
-
-# The stage0 configure step configures the tree to build pretty much the
-# bare minimum that we need for building stage1.
-# Currently the middle end for stage0 will match the selected middle end
-# for the Flambda backend compiler, except when using Flambda 2, when it
-# will be Closure (see configure.ac).
-_build0/config.status: ocaml/configure.ac
-	rm -rf _build0
-	mkdir _build0
-	rsync -a $$(pwd)/ocaml/ $$(pwd)/_build0
-	(cd _build0 && \
-	  cat ../configure_opts_stage0 | tr -d '\n' | xargs -0 ./configure -C \
-	    --prefix=$(stage0_prefix) \
-	    --disable-ocamldoc \
-	    --disable-ocamltest \
-	    --disable-debug-runtime \
-	    --disable-instrumented-runtime \
-	    --disable-debugger)
-
-# stage1 has already been configured by running the configure script.
-# It is configured according to any options requested by the user, including
-# selection of the middle end, but the prefix is set to a temporary build
-# directory.
-
-# This configures stage2 to have the same configure options as stage1
-# except that the prefix is set to the ultimate installation directory.
-# We save the config.status file (which is executable) for fast reconfiguration
-# of the ocaml/ subdirectory during the dune builds for stage1 and stage2.
-# The stage2 configure can be run by make in parallel with that for stage0.
-# We add --enable-ocamltest so that the config.status generated here can be
-# immediately reused for the "compare" target (see below).
-ocaml-stage2-config.status: ocaml/configure.ac
-	rm -rf _stage2_configure
-	mkdir _stage2_configure
-	rsync -a --filter=':- $$(pwd)/ocaml/.gitignore' \
-	  $$(pwd)/ocaml/ $$(pwd)/_stage2_configure
-	(cd _stage2_configure && \
-	  cat ../configure_opts | xargs -0 ./configure -C \
-	    --prefix=$(prefix) \
-	    --enable-ocamltest \
-	    --disable-ocamldoc && \
-	  cp config.status ../ocaml-stage2-config.status)
+install_prefix = @prefix@
+dune = @dune@
+middle_end = @middle_end@
+SHELL = /bin/bash
+
+include ocaml/Makefile.config
+export ARCH
+
+ws_boot   = --root=. --workspace=duneconf/boot.ws
+ws_runstd = --root=. --workspace=duneconf/runtime_stdlib.ws
+ws_main   = --root=. --workspace=duneconf/main.ws
+
+define dune_boot_context
+(lang dune 2.8)
+; We need to call the boot context "default" so that dune selects it for merlin
```

Comment:
Have you tried calling it "boot" but adding a `(merlin)` declaration?

Reply:
I think I did but dune didn't like it for some reason! (Honestly can't remember why, might be worth trying again)

Reply:
Oh, heh - looks like I already tried it in #361. Apparently Dune only handles `(merlin)` well on `opam` stanzas. Probably a bug, but not an urgent one since we can just use `default`.

Reply:
Ah yes, that was it! Thanks for investigating this.